HL-343 | Browser tests

Nizar-Rahme opened the pull request:
## Description :sparkles:
- Implement e2e tests for the applicant side

## Issues :bug:

## Testing :alembic:

## Screenshots :camera_flash:

## Additional notes :spiral_notepad:

nenonja commented on frontend/benefit/applicant/browser-tests/roles/company.ts:
```diff
@@ -0,0 +1,18 @@
+
+const url = getFrontendUrl('/');
+
+const company = Role(url, async (t) => {
```
Role seems to be a clean way to do the logging! I would suggest you to use also MOCK_FLAG (env variable) to completely mock the login from backend, it makes testing and developing easier. Not sure does your backend support it however...
Perhaps I could start using this roles as well.

Nizar-Rahme replied:
I added the MOCK flag and disabled the authentication, it worked well in the local env, but I am getting CORS errors I couldn't manage to work around.

nenonja commented on frontend/package.json:
```diff
@@ -33,6 +33,8 @@
     "bf-applicant:test": "yarn --cwd benefit/applicant test",
     "bf-handler:test": "yarn --cwd benefit/bf-handler test",
+    "bf-applicant:browser-test": "yarn --cwd benefit/applicant browser-test",
+    "bf-applicant:browser-test:ci": "yarn --cwd benefit/applicant browser-test:ci",
```
you probably have to add some certificate flags to make the tests work on review environment. See the same script on kesaseteli-employer

Nizar-Rahme replied:
Copy-pasted from Kesäseteli 🙈

nenonja commented on .github/workflows/bf-review.yml:
```diff
@@ -152,7 +152,26 @@ jobs:
+    steps:
+      - name: Run Acceptance Tests for ${{ matrix.service }}
+        id: testcafe
+        run: yarn --cwd ${{ matrix.dir }} browser-test:ci
```
the script seems to be missing from benefit/applicat package.json

nenonja reviewed (commented):
Great start and this class approach gave me some good ideas too how to implement component apis! I assume that browser tests are nwo working locally, but making them to work in review environment needs some work too..

Nizar-Rahme commented:
For some reason the pipeline is always failing here, I tried to open a new PR with the same changes and the error was gone 🤷 .

Nizar-Rahme commented:
Addressed some of the comments in https://github.com/City-of-Helsinki/yjdh/pull/580
Closing thin PR in favor of https://github.com/City-of-Helsinki/yjdh/pull/580
